apps/agenda/agenda.js

```diff
@@ -30,11 +30,16 @@ var settings = require("Storage").readJSON("agenda.settings.json",true)||{};
 
 CALENDAR=CALENDAR.sort((a,b)=>a.timestamp - b.timestamp);
 
-function getDate(timestamp) {
-  return new Date(timestamp*1000);
+function getDate(timestamp, allDay) {
+  // All day events are always in UTC and always start at 00:00:00, so we
+  // need to "undo" the timezone offsetting to make sure that the day is
+  // correct.
+  var offset = allDay ? new Date().getTimezoneOffset() * 60 : 0
+  return new Date((timestamp+offset)*1000);
 }
-function formatDay(date) {
-  let formattedDate = Locale.dow(date,1) + " " + Locale.date(date).replace(/\d\d\d\d/,"");
+
+function formatDay(date, allDay) {
```

Comment:
`allDay` seems unused in this function, if you really don't need it, let's remove it from the call as well
```suggestion
function formatDay(date) {
```

Reply:
Thanks for catching this, I changed where the logic was a couple times and this is a remnant of that!